### ISSUE TITLE
Extension doesn't load babel config files with babel-eslint

### ISSUE THREAD

**Issue by Nicholaiii** [standard/vscode-standard#100]
**What version of this package are you using?**
1.2.3
**What operating system, Node.js, and npm version?**
W10, 12.7.0, 6.10.0
**What happened?**
Doesn't load babel config when using babel-eslint
`Parsing error: No Babel config file detected for filename`
![image](https://user-images.githubusercontent.com/7280931/73199891-03710780-4136-11ea-89d8-a76e686074bd.png)

**What did you expect to happen?**
Parsing works in terminal using `standard` and it should in vscode as well.

**Comment by RanjithNair** [comment]
Any workaround for this?

**Comment by Nicholaiii** [comment]
> Any workaround for this?

Not that I know of. I stopped using it. I'm using ESLint with eslint-config-standard and a few others (unicorn and jsdoc)

**Comment by RanjithNair** [comment]
One way i got it working is by using a lower version of babel-eslint package. I was on the below version which was causing this issue :-
`"babel-eslint": "^11.0.0-beta.2"`

But once i installed ver 10.0.3, this issue got resolved.

**Comment by FlyKasual** [comment]
I still have this issue, even with a babel.config.json in the root of my project. Here's what it looks like:
```
{
    "presets": [
        [
            "@babel/preset-env",
            {
                "modules": "auto",
                "useBuiltIns": "usage",
                "corejs": 3
            }
        ],
        "@babel/preset-react"
    ],
    "plugins": [
        "@babel/plugin-proposal-class-properties",
        "@babel/plugin-proposal-private-property-in-object"
    ]
}
```
and this is the relevant part of my package.json:
```
    "standard": {
        "parser": "@babel/eslint-parser"
    },
    "devDependencies": {
        "@babel/core": "^7.12.10",
        "@babel/eslint-parser": "^7.12.13",
        "@babel/plugin-proposal-class-properties": "^7.12.1",
        "@babel/plugin-proposal-private-property-in-object": "^7.12.1",
        "@babel/preset-env": "^7.12.10",
        "@babel/preset-react": "^7.12.13",
        "eslint": "^7.19.0",
        "react": "^17.0.1",
        "react-dom": "^17.0.1",
        "standard": "^16.0.3",
        // ...
    }
```
What am I missing here?

**Comment by theoludwig** [comment]
I don't know if it is related but you should not use `babel-eslint`, instead you should use `@babel/eslint-parser`.
See: https://github.com/standard/standard/issues/1601

**Comment by jrtitus** [comment]
I got this same parsing error because I mistakenly opened a parent directory, with directories inside, each using its own Babel configuration. The solution was just to create a new **workspace** and then add the folders to the workspace. VSCode then knows to keep the configurations separate. An obvious oversight on my part, but just thought I'd share in case anyone else tried doing the same thing 😅

**Comment by marcinczenko** [comment]
I wanted to use `@babel/eslint-parser` as this in now the recommended option. It works fine from command line with `yarn standard` but in vscode I keep getting `No Babel config file detected...`. Looks like vscode does some harm to `babelrcRoots` and I still could not find a way to fix it, other than switching back to `babel-eslint`.

**Comment by marcinczenko** [comment]
> I don't know if it is related but you should not use `babel-eslint`, instead you should use `@babel/eslint-parser`.
> See: [standard/standard#1601](https://github.com/standard/standard/issues/1601)

That does not seem to work for me. I checked it in a monorepo and in a "simple" flat project with one `package.json` and one`babel.config.js` in the root. Works from command line, but fails in VSCode.

**Comment by aparajita** [comment]
@marcinczenko The solution for me (I'm using a monorepo) was to specify the path to the root `babel.config.json`:

`.eslintrc.js`
```js
const path = require('path')

module.exports = {
  root: true,
  parser: '@babel/eslint-parser',
  parserOptions: {
    sourceType: 'module',
    babelOptions: {
      configFile: path.join(__dirname, 'babel.config.json')
    }
  },
  ...
}
```

**Comment by marcinczenko** [comment]
@aparajita: Thanks for your suggestion. I do not really want to use `.eslintrc.js` - in the end this is why I am using `standard` in the first place. And then adding `parserOptions` to my `standard` config in my `package.json` does not look like a valid option as `configFile` is only allowed in root programmatic options.

**Comment by Nicholaiii** [comment]
> @aparajita: Thanks for your suggestion. I do not really want to use `.eslintrc.js` - in the end this is why I am using `standard` in the first place. And then adding `parserOptions` to my `standard` config in my `package.json` does not look like a valid option as `configFile` is only allowed in root programmatic options.

There is no standard without using eslint
https://github.com/standard/standard/blob/master/package.json#L17

**Comment by marcinczenko** [comment]
> There is no standard without using eslint
> https://github.com/standard/standard/blob/master/package.json#L17

I am sorry, I do not understand what you want to say. It is (kind of) obvious that standard uses eslint under the hood. So, do you want to say that using `.eslintrc.js` with `standard` is something you would recommend? How does it match the __No Configuration__ principle from https://standardjs.com/#why-should-i-use-javascript-standard-style? Or, did I get everything wrong.

Anyway, as indicated, everything works just fine (and I believe as intended) without `.eslintrc.js` when used outside of the vscode. It also works good (even with vscode) when using `babel-eslint` as the parser. Thus, it looks like there is something happening on the interface between `@babel/eslint-parser` and `vscode-standard`.

It would be nice to know what makes it hard to fix the problem. Is it difficult to replicate? Is it a low priority issue? The solution is blocked by `@babel/eslint-parser`?

**Comment by vladbars** [comment]
Still _No Babel config file detected_ in VS Code when use @babel/eslint-parser. Solutions from here are not working for me

**Comment by mifi** [comment]
For me what did the trick was to add the path of the project I want to lint to `eslint.workingDirectories` in `.vscode/settings.json` of the root of the workspace.
EDIT: This fix was for vscode-eslint, not vscode-standard (wrong repo)

**Comment by marcinczenko** [comment]
@mifi Thanks for your input. The option `eslint.workingDirectories` does not exist on `vscode-standard` extenssion. There is a similar option available, `standard.workingDirectories`, but setting it to point to the project root folder does not help. To have `eslint.workingDirectories` option available, one needs to install `eslint` extension, which, again, misses the point of using StandardJS, for which `vscode-standard` should just act as a dummy facilitator for vscode without introducing its own semantics or philosophy. When you use StandardJS in your project you should not need any extra eslint configuration, besides what StandardJS provides as public settings.

Anyway to check if your workaround does anything in my setup I have installed `eslint` extension and I've set the `eslint.workingDirectories` to also point to my project root folder. As expected, this does not work either.

**Comment by mifi** [comment]
Whoops, my bad. I got the same error but apparently I'm in the wrong github project here because I'm not using vscode-standard

**Comment by RJForgie** [comment]
I'm running into the same issue as @marcinczenko and @vladbars. No Babel config file detected in VS Code when we use @babel/eslint-parser with Standard.

Is this something that will be looked at in the near future?

**Comment by gurpreetsingh41333** [comment]
> For me what did the trick was to add the path of the project I want to lint to `eslint.workingDirectories` in `.vscode/settings.json` of the root of the workspace.
> EDIT: This fix was for vscode-eslint, not vscode-standard (wrong repo)

adding the below snippet in the workspace settings worked for me

 "eslint.workingDirectories": [
			"put your .eslintrc relative path here"
		]

**Comment by feross** [comment]
I believe I've fixed this issue with this PR: https://github.com/standard/vscode-standard/pull/207

If you're using `standard` or `standardx`, etc. with `@babel/eslint-parser` you're `babel.config.js` or `babelrc.js` file should be automatically detected with that PR.

If you're using `standard` with `@babel/eslint-parser` but don't have a custom babel config, you'll still need to add a dummy `babel.config.js` file since that's now required by `@babel/eslint-parser`.

@Divlo can you review and release this as soon as you get the chance?

**Comment by feross** [comment]
Okay, the fix is released in version 1.5.1 of the extension. Please make sure you're using the new extension (https://marketplace.visualstudio.com/items?itemName=standard.vscode-standard) and not the old one. (We changed the owner organization to `standard`, so you'll need to reinstall.
